Userland/Libraries/LibWeb/DOM/Document.h

```diff
@@ -746,7 +744,6 @@ class Document
     bool m_editable { false };
 
     JS::GCPtr<Element> m_focused_element;
-    JS::GCPtr<Element> m_active_element;
```

Comment:
was this cached element purely an artifact of the previous ad hoc implementation? or is there actually a reasonable way to cache this to avoid walking the document or shadow parent tree when we need it?

Reply:
Indeed, it could probably be calculated only when the focused element changes, though I'm not sure what would happen if the focused element were deleted or got moved out of a shadow tree somehow.

Reply:
The main thing the `activeElement` getter does anyway is check if the focused area is the document, and give you the body instead. We don't have a focused area, so if you click off something it actually gives you the `<html>` element.

https://github.com/LadybirdBrowser/ladybird/blob/0140e5e9cccd2c10d5b0ba480f7575a9876c9ee1/Userland/Libraries/LibWeb/HTML/Focus.cpp#L95-L100

`focused_element` should be more like `focused_node`, and allowed to be the document, making sure that this doesn't break any usages, but I felt like that belongs in another PR.